### PR TITLE
Add java formatting support from Maven, reformat code

### PR DESCRIPTION
**Pull request by kkrugler**
With this, you can run `mvn java-formatter:format` to apply the standard Scale Unlimited formatting (exported from Eclipse, in `docs/eclipse-formatter.xml`) to all of the files.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,6 +31,7 @@
         <javac.src.version>1.8</javac.src.version>
         <javac.target.version>1.8</javac.target.version>
         <scala.binary.version>2.11</scala.binary.version>
+		<maven.build.timestamp.format>yyyy-MM-dd HH:mm:ssZ</maven.build.timestamp.format>
```
Comment (vmagotra) [inline pom.xml]: Why is this needed ?

Comment (kkrugler) [reply pom.xml]: It wasn't needed for this. I saw it in the crawler-commons `pom.xml` file. It seemed useful to set up a "build time" property (see https://rterp.wordpress.com/2012/03/16/stamping-version-number-and-build-time-in-properties-file-with-maven/), though that's something for the future.
